content/blog/2023-01-24-aws/index.Rmd

```diff
@@ -0,0 +1,197 @@
+sudo apt install --no-install-recommends r-base
+```
+
+To check the R version, let's first open R.
+
+```{bash eval=F}
+R
+```
+And then run the following code...
+```{r eval=F}
+R.Version()
+quit()
+```
+
+When this guideline was written, the R version was `4.2.2`.
+
+
+### Install RStudio Server
+
+In order to install RStudio server we first need to get `gdebi-core`, a tool allowing us to install local `deb` packages.
+
+```{bash eval=F}
+sudo apt-get install gdebi-core
+```
+
+Now, let's download and install RStudio server (visit [this page](https://posit.co/download/rstudio-server/) for checking the latest version).
+```{bash eval=F}
+wget https://download2.rstudio.org/server/jammy/amd64/rstudio-server-2022.12.0-353-amd64.deb
+```
+```{bash eval=F}
+sudo gdebi rstudio-server-2022.12.0-353-amd64.deb
+```
+
+Now, RStudio server should be up and running on port `8787`! If you have set the security group for port `8787`, to access RStudio server you just need to copy and paste the Public IPv4 address (you can find by selecting the instance you want to connect to in the Instances page of EC2 dashboard) followed by `:8787`, like this `http://X.X.X.X:8787`.  
+If you want to open RStudio by directly copy-pasting the public IPv4 address in a new browser tab, we need to tell the RStudio server to listen on port `80`.
+
+First we need to set the writing access to the `rserver.conf` file.
+```{bash eval=F}
+sudo chmod a+rw /etc/rstudio/rserver.conf
+```
+Then, we need to add the access to port `80`.
+```{bash eval=F}
+echo 'www-port=80' >> /etc/rstudio/rserver.conf
+```
+Finally, we need to restart the RStudio server.
+```{bash eval=F}
+sudo rstudio-server restart
+```
+Now, RStudio can be accessed from port `80`, so, just by copy-pasting the public IPv4 address in a new browser tab.  
+
+Once we try to access the RStudio server, we can see that it requires a username and a password. By default, all the Ubuntu's user are allowed to access the RStudio server.
+If you want to setup a new user, you can run the following command (obviously change `new_username` with the name of the new user).
+
+```{bash eval=F}
+sudo adduser new_username
+```
+
+
+## Install tidyverse and nlmixr2
+
+Before installing `tidyverse` and `nlmixr2` we need to install a few libraries.  
+First of all, let's install `make` and `cmake`.
+```{bash eval=F}
+sudo apt install make
+sudo snap install cmake --classic
+```
+Then, let's install the following libraries.
+```{bash eval=F}
+sudo apt-get install libcurl4-openssl-dev
+sudo apt-get install libxml2-dev
+sudo apt-get install libmpfr-dev
+sudo apt-get install libgmp-dev
+sudo apt-get install libboost-all-dev
+sudo apt install liblapack-dev
+sudo apt install libopenblas-dev
+sudo apt-get install libjpeg-turbo8-dev
+sudo apt-get install libpng-dev
+```
+**Note**: Apparently, Ubuntu server 22 on AWS does not come with many libraries. We found that the libraries written above are needed for `tidyverse` and `nlmixr2` installation. 

```

Comment:
For a bit more clarity, I think that this line should change to "Ubuntu server 22.04".  I would guess that it would also improve search engine results for this page.

Reply:
Good suggestion! Thanks